matrix-tools/Dockerfile

```diff
@@ -7,10 +7,12 @@ FROM --platform=${BUILDPLATFORM} golang:1.24 AS buildstage
 WORKDIR /app
 COPY . /app
 
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg/mod \
+  go mod download
 
 ARG TARGETOS TARGETARCH
-RUN GOOS=$TARGETOS GOARCH=$TARGETARCH CGO_ENABLED=0 go build -o /app/matrix-tools cmd/main.go
+RUN --mount=type=cache,target=/go/pkg/mod \
```

Comment:
Looking at https://www.reddit.com/r/golang/comments/1f4vvir/caching_go_mod_download_in_docker/ / https://medium.com/@marcin.niemira/optimise-docker-build-for-go-c03d6eb8b4b there appears to be another cache, not just the module cache. Unsure what the default is without `go env -w GOCACHE=<path>`

Reply:
MMmh I followed the example from https://docs.docker.com/guides/bake/#testing-and-linting. 

We might be better with explicit configuration. I'll follow the suggestion from these posts.